webapp/src/auth.ts

```diff
@@ -151,20 +171,32 @@ export async function logout() {
     window.location.href = `${window.location.origin}${window.location.pathname}`;
 }
 
+let initialAuthCheck_: Promise<UserProfile | undefined> = undefined;
 /**
  * Checks to see if we're already logged in by trying to fetch user info from
  * the backend. If we have a valid auth token cookie, it will succeed.
  */
-export async function authCheck() {
-    if (!hasIdentity()) { return; }
+export async function authCheck(): Promise<UserProfile | undefined> {
```

Comment:
@eanders-MS , I've reworked authCheck to only run once using the initialAuthCheck_ promise so that it can be called many times and later calls will just return the first result.

I do a similar thing with fetching user preferences.

Reply:
Seems like a good change.

---
In reply to: [521609168](https://github.com/microsoft/pxt/pull/7556#discussion_r521609168) [](ancestors = 521609168)